cognite_toolkit/_cdf_tk/data_classes/_packages.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Optional, overload
+
+from cognite_toolkit._cdf_tk.data_classes._module_directories import ModuleDirectories
+from cognite_toolkit._cdf_tk.data_classes._module_toml import ModuleToml
+
+if sys.version_info >= (3, 11):
+    import toml
+else:
+    import tomli as toml
+
+
+@dataclass
+class SelectableModule:
+    @property
+    def name(self) -> str:
+        return self.path.name
+
+    @property
+    def title(self) -> str | None:
+        return self.definition.description
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SelectableModule):
+            return NotImplemented
+        return self.name == other.name
+
+    def __hash__(self) -> int:
+        return hash(self.name)
+
+    def __str__(self) -> str:
+        return self.name
+
+    definition: ModuleToml
+    path: Path
+
+
+@dataclass
+class Package:
+    """A package represents a bundle of modules.
+    Args:
+        name: the unique identifier of the package.
+    """
+
+    name: str
+    title: str
+    description: str | None = None
+    modules: list[SelectableModule] = field(default_factory=list)
+
+    @classmethod
+    def load(cls, name: str, package_definition: dict) -> Package:
+        return cls(
+            name=name,
+            title=package_definition["title"],
+            description=package_definition.get("description"),
+        )
+
+
+@dataclass
+class Packages(list, MutableSequence[Package]):
+    @overload
+    def __init__(self, packages: Iterable[Package]) -> None: ...
+
+    @overload
+    def __init__(self) -> None: ...
+
+    def __init__(self, packages: Optional[Iterable[Package]] = None) -> None:
+        super().__init__(packages or [])
+
+    def get_by_name(self, name: str) -> Package:
+        for package in self:
+            if package.name == name:
+                return package
+        raise KeyError(f"Package {name} not found")
+
+    @classmethod
+    def load(
+        cls,
+        path: Path,  # todo: relative to org dir
+    ) -> Packages:
+        """Loads the packages in the source directory.
+
+        Args:
+            modules: The module directories to load the packages from.
+        """
+
+        package_definition_path = path / "package.toml"
+        if not package_definition_path.exists():
+            raise FileNotFoundError(f"Package manifest toml not found at {package_definition_path}")
+        package_definitions = toml.loads(package_definition_path.read_text())["packages"]
+
+        collected: dict[str, Package] = {}
+        for package_name, package_definition in package_definitions.items():
+            if isinstance(package_definition, dict):
+                collected[package_name] = Package.load(package_name, package_definition)
+
+        module_directories = ModuleDirectories.load(path, set())
+        selectable_modules = list(
+            {m for m in (cls.get_module(module.dir, path) for module in module_directories) if m is not None}
```

Comment:
I expected this to be
```suggestion
{module.dir /  "module.toml" for module in module_directories) if (module.dir / "module.dir).exists()}

```

Why are you doing a search for it?

Reply:
because of supermodule/submodule, I "climb" from the leaves to find the manifest.toml.